modules/client/src/types.ts

```diff
@@ -0,0 +1,376 @@
+  spender: string;
+  amount: bigint;
+  tokenAddress: string;
+};
+
+export enum SetAllowanceSteps {
+  SETTING_ALLOWANCE = "settingAllowance",
+  ALLOWANCE_SET = "allowanceSet",
+}
+
+export type SetAllowanceStepValue =
+  | { key: SetAllowanceSteps.SETTING_ALLOWANCE; txHash: string }
+  | { key: SetAllowanceSteps.ALLOWANCE_SET; allowance: bigint };
+
+/* Uninstallation */
+export type PrepareUninstallationParams = {
+  daoAddressOrEns: string;
+  pluginAddress: string;
+  pluginInstallationIndex?: number;
+  uninstallationParams?: any[];
+  uninstallationAbi?: string[];
+};
+export enum PrepareUninstallationSteps {
+  PREPARING = "preparing",
+  DONE = "done",
+}
+export type PrepareUninstallationStepValue =
+  | { key: PrepareUninstallationSteps.PREPARING; txHash: string }
+  | {
+    key: PrepareUninstallationSteps.DONE;
+  } & ApplyUninstallationParams;
+
+export type ApplyUninstallationParams = ApplyInstallationParamsBase;
+export type DecodedApplyUninstallationParams = ApplyInstallationParamsBase;
+
+/* Permissions */
+type PermissionParamsBase = {
+  where: string;
+  who: string;
+  permission: string;
+};
+type PermissionDecodedParamsBase = PermissionParamsBase & {
+  permissionId: string;
+};
+export type GrantPermissionParams = PermissionParamsBase;
+export type RevokePermissionParams = PermissionParamsBase;
+export type GrantPermissionDecodedParams = PermissionDecodedParamsBase;
+export type RevokePermissionDecodedParams = PermissionDecodedParamsBase;
+
+export type GrantPermissionWithConditionParams = PermissionParamsBase & {
+  condition: string;
+};
+export type GrantPermissionWithConditionDecodedParams = PermissionParamsBase & {
+  condition: string;
+  permissionId: string;
+};
+
+export type HasPermissionParams = PermissionParamsBase & {
+  daoAddressOrEns: string;
+  data?: Uint8Array;
+};
+
+const Permissions = {
+  UPGRADE_PERMISSION: "UPGRADE_PERMISSION",
+  SET_METADATA_PERMISSION: "SET_METADATA_PERMISSION",
+  EXECUTE_PERMISSION: "EXECUTE_PERMISSION",
+  WITHDRAW_PERMISSION: "WITHDRAW_PERMISSION",
+  SET_SIGNATURE_VALIDATOR_PERMISSION: "SET_SIGNATURE_VALIDATOR_PERMISSION",
+  SET_TRUSTED_FORWARDER_PERMISSION: "SET_TRUSTED_FORWARDER_PERMISSION",
+  ROOT_PERMISSION: "ROOT_PERMISSION",
+  CREATE_VERSION_PERMISSION: "CREATE_VERSION_PERMISSION",
+  REGISTER_PERMISSION: "REGISTER_PERMISSION",
+  REGISTER_DAO_PERMISSION: "REGISTER_DAO_PERMISSION",
+  REGISTER_ENS_SUBDOMAIN_PERMISSION: "REGISTER_ENS_SUBDOMAIN_PERMISSION",
+  MINT_PERMISSION: "MINT_PERMISSION",
+  MERKLE_MINT_PERMISSION: "MERKLE_MINT_PERMISSION",
+  MODIFY_ALLOWLIST_PERMISSION: "MODIFY_ALLOWLIST_PERMISSION",
+  SET_CONFIGURATION_PERMISSION: "SET_CONFIGURATION_PERMISSION",
+};
+
+const PermissionIds = Object.entries(Permissions).reduce(
+  (acc, [k, v]) => ({ ...acc, [k + "_ID"]: keccak256(toUtf8Bytes(v)) }),
+  {} as { [k: string]: string },
+);
+Object.freeze(Permissions);
+export { Permissions };
+Object.freeze(PermissionIds);
+export { PermissionIds };
```

Comment:
Should this really be part of the types file? 
This feels more like a constant

Reply:
it is, thrue thank you for pointng it out